oximeter/db/tests/integration_test.rs

```diff
@@ -0,0 +1,380 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use anyhow::Context;
+use clickward::{Deployment, KeeperClient, KeeperError};
+use dropshot::test_util::log_prefix_for_test;
+use omicron_test_utils::dev::poll;
+use omicron_test_utils::dev::test_setup_log;
+use oximeter::test_util;
+use oximeter_db::{Client, DbWrite, OxqlResult, Sample, TestDbWrite};
+use std::default::Default;
+use std::time::Duration;
+
+pub struct TestInput {
+    n_projects: usize,
+    n_instances: usize,
+    n_cpus: usize,
+    n_samples: usize,
+}
+
+impl Default for TestInput {
+    fn default() -> Self {
+        TestInput { n_projects: 2, n_instances: 2, n_cpus: 2, n_samples: 50 }
+    }
+}
+
+impl TestInput {
+    fn n_timeseries(&self) -> usize {
+        self.n_projects * self.n_instances * self.n_cpus
+    }
+
+    fn n_points(&self) -> usize {
+        self.n_projects * self.n_instances * self.n_cpus * self.n_samples
+    }
+}
+
+#[tokio::test]
+async fn test_cluster() -> anyhow::Result<()> {
```

Comment:
I see there's only a single test here, is the intention to port all replicated tests in `oximeter/db/src/client/mod.rs` to be tested against clickward eventually? That'd be really nice!

No need to port them all now, just wondering if this is what you had in mind as well. Those tests take forever.

Reply:
> I actually believe pretty strongly that it's unnecessary to port those tests, and that the duplicates running against both single and replicated test cases can be safely moved to run solely against a single node.

The replicated tests came to be as inconsistencies between the single node SQL init file and replicated one started popping up and weren't being caught by any test. I'd be a bit wary of removing them altogether until we have a better way of ensuring those two files stay in sync.

> If we eventually move over to an all distributed / replicated configuration and get rid of single node altogether, then we can always spin up a cluster configuration with a single replica and a single keeper and test that way with all our existing tests rather than use single node mode as long as the runtime cost is kept to a minimum.

This could definitely work! Since the SQL to initialise a single-node setup is different a to replicated one I'm very much in favour of removing single-node functionality eventually.

Reply:
I just added a test to ensure that schema files are disjoint for the replicated files. I'll add another one that ensures the replicated schema matches the single node schema. That should be enough ensure that they stay in sync. We can then remove the duplicate replicated tests safely in a follow up.

Reply:
> I just added a test to ensure that schema files are disjoint for the replicated files. I'll add another one that ensures the replicated schema matches the single node schema. That should be enough ensure that they stay in sync. We can then remove the duplicate replicated tests safely in a follow up.

I realized that for this test to do what you want, I'll also have to check that the columns and their types per table are identical. I'll take a stab at that as well.

Reply:
I think you could probably get away with checking that their create-table statements are the same, just comparing them as strings. You can also select the data as JSON, and then rely on `serde_json::Value::partial_cmp()` to do the right thing.

Reply:
> I think you could probably get away with checking that their create-table statements are the same, just comparing them as strings. You can also select the data as JSON, and then rely on `serde_json::Value::partial_cmp()` to do the right thing.

I think this is somewhat tricky because of the `_local table prefixes` and `On Cluster. There needs to be some string filtering. I also realized that the table engines types also need to be compared such that `ReplacingMergeTree` matches `ReplicatedReplacingMergeTree`, etc.. Then we also need to ensure that the order by matches and TTL matches. 

Over all this is a huge pain in the ass. I might punt on this right now.

Reply:
Sounds good! It's not that urgent anyway :)